iocore/dns/SRV.h

```diff
@@ -50,9 +49,8 @@ operator<(const SRV &left, const SRV &right)
 }
 
 struct SRVHosts {
-  unsigned int srv_host_count   = 0;
   unsigned int srv_hosts_length = 0;
-  SRV hosts[HOST_DB_MAX_ROUND_ROBIN_INFO];
+  std::vector<SRV> hosts;
```

Comment:
If we remove `srv_hosts_length`, we need to loop over every time we need it. Do we really want to do that?